src/app/server/java/ChipThreadWork.cpp

```diff
@@ -26,12 +28,23 @@ namespace {
 struct WorkData
 {
     WorkCallback callback;
-    sem_t done;
+    std::mutex mux;
+    std::condition_variable cond;
+    bool done = false;
 
-    WorkData() { sem_init(&done, 0 /* shared */, 0); }
-    ~WorkData() { sem_destroy(&done); }
-    void Post() { sem_post(&done); }
-    void Wait() { sem_wait(&done); }
+    WorkData() {}
```

Comment:
Why not implicit constructor/destructor (or ` = default`)? The same thing below.